app/controllers/dictionary_controller.rb

```diff
@@ -0,0 +1,15 @@
+require 'json'
+
+class DictionaryController < ApplicationController
+  skip_before_action :authenticate_user!, only: [:search]
+  def search
+    @search_input = params[:input]
+    # TODO: ensure that @search_input is properly sanitized
+    @dictionary_output = Dictionary.find_by key: @search_input
+    if @dictionary_output.nil?
+      render json: JSON.generate({ description: 'not found', id: nil })
```

Comment:
Check out active model serializers to serialize your JSON. It is what everyone uses at CMM for the most part